### PR TITLE
refactor: columnpicker remove hard coded value

### PR DESCRIPTION
**Pull request by JComins000**
## Description

I saw a hardcoded value used in two different places. Here's the location where it's already being defined and used.
https://github.com/determined-ai/determined/blob/main/webui/react/src/pages/F_ExpList/F_ExperimentList.tsx#L1073-L1077

## Test Plan

Make sure the column picker looks the same as before
image

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/ColumnPickerMenu.tsx [hunk 1]
```diff
@@ -284,7 +280,7 @@ const ColumnPickerMenu: React.FC<ColumnMenuProps> = ({
                   ),
                   forceRender: true,
                   key: canonicalTab,
-                  label: locationLabelMap[canonicalTab],
+                  label: locationLabelMap[canonicalTab as keyof typeof locationLabelMap],
                 };
               })}
             />
```
Comment (JComins000) [inline webui/react/src/components/ColumnPickerMenu.tsx:286]: tbh i'm not really sure why we have a `tabs.length === 1` down two lines. are we planning on using that condition later?

Comment (johnkim-det) [reply webui/react/src/components/ColumnPickerMenu.tsx:286]: the Searches view is a case where we only display columns with location `EXPERIMENT`, so we don't need the `Pivot` component and can just display one `ColumnPickerTab` instead.

**Review by johnkim-det — APPROVED** [review]
LGTM

File: webui/react/src/components/ColumnPickerMenu.tsx (hunk 1 above)
Comment (JComins000) [reply webui/react/src/components/ColumnPickerMenu.tsx:286]: do we ever do that anywhere?

Comment (JComins000) [reply webui/react/src/components/ColumnPickerMenu.tsx:286]: looks like there's only one reference right now
image

**Pull request merged**